docs/oauth_provider.md

```diff
@@ -0,0 +1,24 @@
+## Introduction
+Starting with version 2.0.1, this plugin allows Jenkins to act as an OAuth provider for Bitbucket Server. Bitbucket Server admins can now
+add Jenkins as an Application Link, allowing users to run various Jenkins operations from within Bitbucket Server.   
```

Comment:
The [screen shot](https://github.com/jenkinsci/atlassian-bitbucket-server-integration-plugin/blob/e5aefd493d5fd6429e40106da272cb9eaa2f92ac/docs/authorize.png) says: "By allowing this, you can use **Bitbucket** without storing your Bitbucket password in **Jenkins**." That reads like Bitbucket Server will then trust Jenkins.

Here though, you write about running "Jenkins operations from within Bitbucket Server", as if Jenkins is trusting Bitbucket Server. Could you clarify which is trusting which?

I'd like to understand how much this feature increases the risk to Bitbucket Server in case Jenkins is compromised due to a vulnerability in some Jenkins plugin.

Reply:
Thanks for pointing this out. Here, Jenkins is trusting Bitbucket Server. I have fixed the screen shot here - https://github.com/jenkinsci/atlassian-bitbucket-server-integration-plugin/pull/186. We will not be storing any BB Server password or access token in Jenkins server.